go/websocketbridge/websocket_bridge_test.go

```diff
@@ -0,0 +1,202 @@
+	"github.com/Originate/exocom/go/structs"
+	"github.com/Originate/exocom/go/websocketbridge"
+
+	"golang.org/x/net/websocket"
+)
+
+func newExocom(port int) *exocomMock.ExoComMock {
+	exocom := exocomMock.New()
+	go func() {
+		err := exocom.Listen(port)
+		if err != nil && err != http.ErrServerClosed {
+			log.Fatal(err)
+		}
+	}()
+	return exocom
+}
+
+// Cucumber step definitions
+// nolint gocyclo
+func FeatureContext(s *godog.Suite) {
+	var exocom *exocomMock.ExoComMock
+	var websocketBridgeInstance *websocketbridge.WebsocketBridge
+	var ws *websocket.Conn
+	exoPort := 4100
+	clientPort := 5435
+	var outgoingMessageId string
+	var savedError error
+
+	s.BeforeSuite(func() {
+		exocom = newExocom(exoPort)
+	})
+
+	s.BeforeScenario(func(interface{}) {
+		outgoingMessageId = ""
+		savedError = nil
+		ws = nil
+	})
+
+	s.AfterScenario(func(interface{}, error) {
+		exocom.Reset()
+		err := websocketBridgeInstance.Reset()
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	s.AfterSuite(func() {
+		err := exocom.Close()
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	s.Step(`a websocket bridge connected to Exocom`, func() error {
+		websocketBridgeInstance = &websocketbridge.WebsocketBridge{
+			ClientPort: clientPort,
+			Config: websocketbridge.Config{
+				Host: "localhost",
+				Port: exoPort,
+				Role: "websocket-test",
+			},
+		}
+		err := websocketBridgeInstance.ConnectToExocom()
+		if err != nil {
+			return err
+		}
+		err = websocketBridgeInstance.ListenForClients()
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	s.Step(`receiving this message from the client:`, func(payloadStr *gherkin.DocString) error {
+		origin := "http://localhost/"
+		url := fmt.Sprintf("ws://localhost:%v/ws", clientPort)
+		var err error
+		ws, err = websocket.Dial(url, "", origin)
+		if err != nil {
+			log.Fatal(err)
+		}
+		var message structs.Message
+		err = json.Unmarshal([]byte(payloadStr.Content), &message)
+		if err != nil {
+			log.Fatal(err)
+		}
+		outgoingMessageId = message.ID
+		marshaledMessage, err := json.Marshal(message)
+		if _, err := ws.Write(marshaledMessage); err != nil {
+			log.Fatal(err)
+		}
+		return err
+	})
+
+	s.Step(`receiving this message from exocom:`, func(payloadStr *gherkin.DocString) error {
+		var message structs.Message
+		err := json.Unmarshal([]byte(payloadStr.Content), &message)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = exocom.WaitForConnection()
+		if err != nil {
+			return err
+		}
+		for key := range websocketBridgeInstance.SocketMap {
+			message.SessionID = key
+		}
+		return exocom.Send(message)
+	})
+
+	s.Step(`websocket bridge makes the websocket request:`, func(payloadStr *gherkin.DocString) error {
+		t := template.New("request")
+		t, err := t.Parse(payloadStr.Content)
+		if err != nil {
+			return err
+		}
+		var expectedMessageBuffer bytes.Buffer
+		err = t.Execute(&expectedMessageBuffer, map[string]interface{}{"outgoingMessageId": outgoingMessageId})
```

Comment:
You don't appear to be using the template / outgoingMessageId logic, so it can be removed.